hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -151,7 +152,7 @@ public HoodieWriteClient(JavaSparkContext jsc, HoodieWriteConfig clientConfig,
               config.getUpsertShuffleParallelism());
 
       // perform index loop up to get existing location of records
-      JavaRDD<HoodieRecord<T>> taggedRecords = index.tagLocation(dedupedRecords, table);
+      JavaRDD<HoodieRecord<T>> taggedRecords = getIndex().tagLocation(dedupedRecords, table);
```

Comment:
lets remove `index` => `getIndex()`  refactors. its still a private variable that can be accessed within the same class.

Reply:
Sure, made a package private constructor for testing purpose to inject a mock index.